cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -163,27 +162,41 @@ public void createTethering(FullHttpRequest request, HttpResponder responder, @P
       throw e;
     }
 
-    // We don't need to keep track of the client metadata on the server side.
-    PeerMetadata peerMetadata = new PeerMetadata(tetherRequest.getNamespaceAllocations(), Collections.emptyMap(),
-                                                 tetherRequest.getDescription());
-    // We don't store the peer endpoint on the server side because the connection is initiated by the client.
-    PeerInfo peerInfo = new PeerInfo(peer, null, TetheringStatus.PENDING, peerMetadata,
-                                     tetherRequest.getRequestTime());
     try {
-      store.addPeer(peerInfo);
-    } catch (PeerAlreadyExistsException pae) {
-      // Peer is already configured, treat this as a no-op.
-      responder.sendStatus(HttpResponseStatus.OK);
-      return;
-    } catch (Exception e) {
+      String content = request.content().toString(StandardCharsets.UTF_8);
+      TetheringConnectionRequest tetherRequest = GSON.fromJson(content, TetheringConnectionRequest.class);
+
+      // We don't need to keep track of the client metadata on the server side.
+      PeerMetadata peerMetadata = new PeerMetadata(tetherRequest.getNamespaceAllocations(), Collections.emptyMap(),
+                                                   tetherRequest.getDescription());
+      try {
+        PeerInfo existingPeerInfo = store.getPeer(peer);
+        if (existingPeerInfo.getTetheringStatus() == TetheringStatus.REJECTED) {
```

Comment:
I thought we had decided that rejecting a request would delete it?

Reply:
I'll do that in a separate PR as it requires coordination with UI change  - currently UI makes a backend calls to reject and delete when the user clicks on Reject in the UI.

Reply:
neverind, fixed in this PR